### PR TITLE
Updated Makefile and Documentation

### PR DESCRIPTION
**Pull request by benglines**
I changed the Makefile to require the user to specify the part name rather than expecting the user to only use the xc7a35tcsg324-1 part or to go into the Makefile to change the part name.

I also changed the Makefile to generate an .html page with a "frame view" of a bitstream. The feature was already implemented, but the Makefile made no use of it. Now it will be generated by default.

I updated the documentation to be more clear of the different features included in this tool. I also updated it to reflect changes in the Makefile that require the user to specify the part name. I also Changed the README from Markdown to the reStructuredText format that seems to be used in the majority of SymbiFlow repos.

### CODE REVIEW

**Comment by mithro** [comment]
@kgugala Can you find someone to review this?

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,6 @@ clean::
 
 # -------------------- BACKEND --------------------
 
-PART = xc7a35tcsg324-1
-#PART = xc7a50tfgg484-1
-
```
Comment (ghost) [inline Makefile:42]: I would rather keep default part in makefile. You can remove commented-out line and move default one to the top of the file and then change assign operator to `?=` like this: `PART ?= xc7a35tcsg324-1`
(reference https://www.gnu.org/software/make/manual/html_node/Setting.html)

File: Makefile [hunk 2]
```diff
@@ -75,5 +72,7 @@ bitstreamData.json: bitstream.bits bithtml.py grid/.dir frames/.dir
 		--bits=bitstream.bits \
 		--dump-grid=$@ --grid-dir=$$PWD/grid
 
+all:: bitstream.html
+
```
Comment (ghost) [inline Makefile]: I can't see any point in generating `bitstream.html` by default. It's not used anymore and it was left just for reference. In fact it should be removed (with the corresponding code in `bithtml.py`) or cleaned and left as optional target.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,24 @@
+================
+Bitstream Viewer
+================
+
+This tool has two primary functions:
```
Comment (ghost) [inline README.rst:7]: Please left the original info from README.md:
`The purpose of this tool is to visualize the bits and their corresponding tags in a given bitstream.`

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+1. Tile View: A grid of tiles will be generated to display the features in a bitstream. Each tile that is inhabited by features will be highlighted. By clicking on a highlighted tile, a pop-up will appear with the features within that tile and their corresponding bits. This is accomplished with `Vue.js, <https://vuejs.org/>`_ a JavaScript framework.
```
Comment (ghost) [inline README.rst:9]: Like this description :+1:

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,24 @@
+================
+Bitstream Viewer
+================
+
+This tool has two primary functions:
+
+1. Tile View: A grid of tiles will be generated to display the features in a bitstream. Each tile that is inhabited by features will be highlighted. By clicking on a highlighted tile, a pop-up will appear with the features within that tile and their corresponding bits. This is accomplished with `Vue.js, <https://vuejs.org/>`_ a JavaScript framework. 
+
+2. Frame View: A grid of frames will be generated to display features in a bitstream. Each frame that contains bits associated with a feature used by the bitstream will be highlighted. By clicking on a highlighted frame, the page will be redirected to a grid of all the bits for that specific frame.
```
Comment (ghost) [inline README.rst]: Point that this is an optional function

**Review by ghost — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (benglines) [reply Makefile:42]: Okay! I put in a line for the default part as you suggested.

File: Makefile (hunk 2 above)
Comment (benglines) [reply Makefile]: I changed the file to not have bitstream.html build by default

File: README.rst (hunk 3 above)
Comment (benglines) [reply README.rst:7]: I put that description back into the file

File: README.rst (hunk 5 above)
Comment (benglines) [reply README.rst]: I added a line to show that it is an optional function

**Review by ghost — APPROVED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
Looks good. One minor thing. Can you sign your commits (-s)?

**Comment by ghost** [comment]
@benglines please make sure to signoff your commits as without CI fails at the DCO check. Once you fix it we can merge the PR.

**Comment by benglines** [comment]
Sorry I did not do this earlier. I believe everything should be in order
now. Let me know if it's still not right

**Comment by ghost** [comment]
@benglines Can add empty line between commit description and Signed-off-by in https://github.com/SymbiFlow/symbiflow-bitstream-viewer/pull/10/commits/54f856c0fb7d531f0c38c6195bbb96ad18f1fd3e to satisfy DCO? Thanks!

**Comment by benglines** [comment]
Done!

**Review by ghost — APPROVED** [review]
Thanks!

**Pull request merged**